File: xdm-core/src/main/java/xdm/core/media/muxer/Muxer.java
package xdm.core.media.muxer;

import java.util.List;
import java.util.function.IntConsumer;

public interface Muxer {

  boolean mux(
      List<String> segments, String outputFile, IntConsumer progressCallback, String tempDir);

  boolean mux(
      String file1, String file2, String outputFile, IntConsumer progressCallback, String tempDir);

  boolean mux(
      List<String> audioSegments,
      List<String> videoSegments,
      String outputFile,
      IntConsumer progressCallback,
      String tempDir);

  void stop();
}

File: xdm-core/src/main/java/xdm/core/media/muxer/Muxer.kt
package xdm.core.media.muxer


interface Muxer {
    fun mux(
        segments: List<String>, outputFile: String, progressCallback: (Int) -> Unit, tempDir: String,
        independentSegement: Boolean,
        isMp4: Boolean,
        // Manifest signalled a discontinuity (e.g. EXT-X-DISCONTINUITY): repair the output timeline
        // across mid-stream timestamp resets. No-op when false.
        discontinuous: Boolean = false,
    ): Boolean

    fun mux(
        file1: String, file2: String, outputFile: String, progressCallback: (Int) -> Unit, tempDir: String
    ): Boolean

    fun mux(
        audioSegments: List<String>,
        videoSegments: List<String>,
        outputFile: String,
        progressCallback: (Int) -> Unit,
        tempDir: String,
        independentSegement: Boolean,
        isMp4: Boolean,
        discontinuous: Boolean = false,
    ): Boolean

    fun stop()
}
